pkg/graphql/federation/node_extractor.go

```diff
@@ -0,0 +1,194 @@
+package federation
+
+import (
+	"github.com/jensneuse/graphql-go-tools/internal/pkg/unsafebytes"
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/plan"
+)
+
+// Extract all fields from Entities.
+// If the Entity is extended - only local fields (fields without external directive).
+
+type nodeExtractor struct {
```

Comment:
I have the feeling that this is the wrong place for the node extractor. Maybe it should live in the `ast` or its own package.